packages/dataviews/src/components/dataviews/index.tsx

```diff
@@ -53,29 +58,42 @@ type DataViewsProps< Item > = {
 	? { getItemId?: ( item: Item ) => string }
 	: { getItemId: ( item: Item ) => string } );
 
+export type DataViewsCompositionProps< Item > = {
```

Comment:
Instead using `Omit` and `Partial`, it may be easier to create a `DataViewsBaseProps` type that includes the mandatory items.
Than create the composition and controlled type like so ( example ):
```
type DataViewsCompositionProps< Item > = {
  children: ReactNode;
} & DataViewsBaseProps< Item >;

type DataViewsControlledProps< Item > = {
  header?: ReactNode;
  // Whatever else is needed for the controlled version.
} & DataViewsBaseProps< Item >;
```

Reply:
Not aligned with the [minimal change](https://github.com/Automattic/wp-calypso/pull/102996#issuecomment-2841332366). We'll change this.

Reply:
> Not aligned with the https://github.com/Automattic/wp-calypso/pull/102996#issuecomment-2841332366. We'll change this.

Having said the above, given we are already renaming this to `DataViewsCompositionProps`, changing it to `DataViewsBaseProps` and removing a couple props, still feels pretty minimal to me :)
But I am ok with sticking with the above.